core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -321,8 +321,11 @@ public T get(Settings settings) {
     public String getRaw(Settings settings) {
         // They're using the setting, so we need to tell them to stop
         if (this.isDeprecated() && this.exists(settings)) {
+            if (deprecationLogger.get() == null) {
```

Comment:
I usually use the initialization-on-demand holder idiom for these scenarios https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom
(sorry if this sounds like bike-shedding)